src/keys.rs

```diff
@@ -40,6 +42,49 @@ pub trait SecretKey: ByteArray + Clone + PartialEq + Eq + Add<Output = Self> + D
 /// implementations need to implement this trait for them to be used in Tari.
 ///
 /// See [SecretKey](trait.SecretKey.html) for an example.
+/// cbindgen:ignore
+#[cfg(feature = "borsh")]
+pub trait PublicKey:
```

Comment:
I think having a different trait depending on whether the borsh feature flag is enabled or not could be troublesome.

Any reason we require these trait bounds in the trait itself?